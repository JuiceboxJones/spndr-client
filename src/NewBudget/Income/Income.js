import React, { Component } from 'react';
import ApiService from '../../services/api-fetch-services';
import HistoryHelper from '../../History/History';


class Income extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }
  
  state = {
    previousIncome: []
  }



  componentDidMount(){
    ApiService.getIncome()
    .then(data => {
      this.setState({ previousIncome: data })
    })
      
  }

  handleSubmitIncome = e => {
    const income = {
      add_savings:  e.target.savings.value,
      income:       e.target.income.value,
      bank_balance: e.target.balance.value
    }
    ApiService.postIncome(income)
    this.handleDeletePrev()
    this.handleNext()

  }

  handleDeletePrev(){
    if(this.state.previousIncome.length !== 0){
      ApiService.deleteIncome(this.state.previousIncome[0].id)}
  }

  handleNext(e){
    HistoryHelper.historyGoTo('/expenses', this.props)
  }

  handlePreviousIncome = () => {
    const income = this.state.previousIncome
    if(income === null || income.length < 1){
      return <div>Let's Get Started! Step 1 of 3</div>
    } else{
      const inc = this.state.previousIncome[0]
      return <div>
        <p>Previous Income:</p>
        <p>Bank balance: ${inc.bank_balance} Income: ${inc.income} Add to savings: ${inc.add_savings}</p>
        <button type='button' onClick={(e) => this.handleNext(e)}>Use previous income</button>
      </div>
    }
  }

  render() { 


    return ( 
      <div>
        <header>
          <h2>Add Income</h2>
        </header>
          <div>
          {this.handlePreviousIncome()}
          </div>
      <form className="income" onSubmit={(e) => this.handleSubmitIncome(e)}>
          <div className="income_section">
            <p>New Income:</p>
            <label htmlFor='user_balance'>Current account balance</label>
            <p/>
            <input type='number' step='0.01' name='balance' id='balance' />
            <label htmlFor="user_income">Monthly Income</label>
            <p/>
            <input type='number' step='0.01' required name="income" id="income" />
            <label htmlFor="user_savings">Amount to add to savings</label>
            <p/>
            <input type='number' step='0.01' required name="savings" id="savings" />
          </div>
          <button type="submit">Create new income</button>
          </form>
          </div>
     );
  }
}
 
export default Income;