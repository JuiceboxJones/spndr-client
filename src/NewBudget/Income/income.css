#header h2 {
  margin-top: 0px;
}

.income_page_container{
  padding: 15px;
}

.income{
  width: 200px;
  margin: 0 auto;
  position: relative;
  top: 30px;
}

.form_box{
  margin: 0 auto;
  width: 33%;
  margin-top: 5%;
  border-radius: 8px;
  background: #f5f5f5;
  padding: 10px;
  height: 51%;
  padding-bottom: 20px;
}

.income label{
  margin: 0 auto;
}
.income h2{
  margin-bottom: 45px;
}

#income_buttons{
  border: none;
  width: 150px;
  height: 25px;
  border-radius: 5px;
  background:#A7C4C0;
  cursor: pointer;
  font-weight: bold;
  margin: 20px;
}

#income_buttons:hover{
  background: #024e2e9c;
  color:white;
}
@media screen and (max-width: 480px) {
  .form_box{
    width: 80%;
  }
  h2{
    font-size: 15px;
  }
  #income_buttons{
    margin-bottom: 0;
    margin-top: 10px;
  }
  input{
    margin: 0;
    margin-bottom: 5px;
  }
  .income_page_container{
    height: 153%;
  }
  .income h2{
    margin-bottom: 23px;

  }
}
@media screen and (max-width: 330px) {
.income_page_container{
  height: 100vh;
  padding: 0;
}
.form_box{
  height: 90%;
}
#income_buttons{
  margin-bottom: -5px;
  margin-top: 7px;
}
}