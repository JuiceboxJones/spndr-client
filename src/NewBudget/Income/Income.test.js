import React from 'react';
import ReactDOM from 'react-dom';
import Income from './Income'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Income />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Income />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});