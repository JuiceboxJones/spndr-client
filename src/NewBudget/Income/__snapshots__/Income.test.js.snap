// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="income_page_container"
>
  <header
    id="header"
  >
    <h2>
      Add Income
    </h2>
    <p>
      Step 2 of 3
    </p>
  </header>
  <div
    className="form_box"
  >
    <div
      className="previous_income"
    >
      <div>
        Let's Get Started! Step 1 of 3
      </div>
    </div>
    <form
      className="income"
      onSubmit={[Function]}
    >
      <div
        className="income_section"
      >
        <h2>
          New Income:
        </h2>
        <label
          htmlFor="user_balance"
        >
          Current account balance
        </label>
        <p />
        <input
          id="balance"
          name="balance"
          step="0.01"
          type="number"
        />
        <label
          htmlFor="user_income"
        >
          Monthly Income
        </label>
        <p />
        <input
          id="income"
          name="income"
          required={true}
          step="0.01"
          type="number"
        />
        <label
          htmlFor="user_savings"
        >
          Amount to add to savings
        </label>
        <p />
        <input
          id="savings"
          name="savings"
          required={true}
          step="0.01"
          type="number"
        />
      </div>
      <button
        id="income_buttons"
        type="submit"
      >
        Create new income
      </button>
    </form>
  </div>
</div>
`;
